### PR TITLE
Snackbar

### PR DESCRIPTION
**Pull request by yschimke**
When I was ripping out raw usage of Toast in my app, I needed something modern android to replace it with.

I didn't want to pull in mobile-compose for Snackbar functionality.

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/UiMessage.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+package com.google.android.horologist.compose.snackbar
+
+import com.google.android.horologist.compose.navscaffold.ExperimentalComposeLayoutApi
+import java.util.UUID
+
+@ExperimentalComposeLayoutApi
+public data class UiMessage(
+    val message: String,
+    val id: String = UUID.randomUUID().toString(),
+    val error: Boolean = true
```
Comment (alexvanyo) [inline compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/UiMessage.kt:26]: I don't think I saw `error` being used anywhere?

File: compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/SnackbarViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import androidx.navigation.NavHostController
+import androidx.wear.compose.material.Scaffold
+import com.google.android.horologist.compose.navscaffold.ExperimentalComposeLayoutApi
+import com.google.android.horologist.compose.snackbar.material.SnackbarDuration
+import com.google.android.horologist.compose.snackbar.material.SnackbarHostState
+import kotlinx.coroutines.launch
+
+/**
+ * A ViewModel that backs the WearNavScaffold to allow each composable to interact and effect
+ * the [Scaffold] positionIndicator, vignette and timeText.
```
Comment (alexvanyo) [inline compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/SnackbarViewModel.kt:32]: Out of date KDoc I believe

File: compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/SnackbarManager.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+package com.google.android.horologist.compose.snackbar
+
+import com.google.android.horologist.compose.navscaffold.ExperimentalComposeLayoutApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+
+@ExperimentalComposeLayoutApi
+public class SnackbarManager {
```
Comment (alexvanyo) [inline compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/SnackbarManager.kt:26]: Optional, but all of `SnackbarManager` could probably be replaced with just a `mutableStateListOf<UiMessage>()`, with `MutableList.removeIf`, `MutableList.add`, and `snapshotFlow {  }` to observe changes.

Should be a little bit cleaner than the `MutableStateFlow`!

File: compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/UiMessage.kt (hunk 1 above)
Comment (yschimke) [reply compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/UiMessage.kt:26]: Will remove.  Generally I'd love any insight into what bits should be here.  I've looked at Tivi and androidx.compose.material.SnackbarHost so maybe I've got a bit of a combined approach.

Should SnackbarViewModel be in this library? Or just SnackbarHostState and DialogSnackbarHost?

Comment (alexvanyo) [reply compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/UiMessage.kt]: That's a good question... I feel like just `SnackbarHostState` and `DialogSnackbarHost`, but I think the `SnackbarViewModel` and `SnackbarManager` (or some combined version of the two) would be good to have in the sample for example usage.

Comment (yschimke) [reply compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/UiMessage.kt]: Done.

File: compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/DialogSnackbarHost.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.wear.compose.material.MaterialTheme
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.material.dialog.Confirmation
+import com.google.android.horologist.compose.navscaffold.ExperimentalComposeLayoutApi
+
+@ExperimentalComposeLayoutApi
+@Composable
+public fun DialogSnackbarHost(
+    modifier: Modifier = Modifier,
+    hostState: SnackbarHostState,
```
Comment (luizgrp) [inline compose-layout/src/main/java/com/google/android/horologist/compose/snackbar/DialogSnackbarHost.kt:34]: nit: should potentially come before the modifier param - not sure if there is an exception for state class

**Review by luizgrp — APPROVED** [review]

**Comment by yschimke** [comment]
Failures is a bug because of tiles minSdk.

**Pull request merged**